src/main/java/io/jenkins/plugins/httpclient/RobustHTTPClient.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.httpclient;
```

Comment:
Package name does differ from artifactId. I would propose `apache.httpcomponents.client.api` or so.
OTOH the code does not use the library on its own. So more reviews may be required

Reply:
That package name is pretty awkward, and also gives the false impression that this is Apache code.

Using the `artifactId` ~ `shortName`, in some modified form, as a final package component is indeed typical. This particular `artifactId` is quite unwieldy though. Also I am not sure there is a solid convention for “miscellaneous source packages added to a library wrapper”, which is not really the same as “main source package of a regular plugin”.

Reply:
I assume Oleg meant `io.jenkins.plugins.apache.httpcomponents.client.api`

Reply:
Normally `-` is converted to `_` or camel casing when constructing a package component from an `artifactId`, so something like `io.jenkins.plugins.apache_httpcomponents_client_4_api`. Pretty ugly.